pkg/signature-verifier/signature-verifier.go

```diff
@@ -27,12 +27,22 @@ import (
 
 type VerifyOptions struct {
 	CosignVerifyOptions
+	NotationVerifyOptions
```

Comment:
What if both fields are set? IMO they should be pointers and check that only one is set.

Reply:
ok, I was wrong. It's possible that a user wants to be able to verify some gadgets with cosign and some with notation, so we should check each gadget against all verication methods available and return if one matches.